src/main/java/com/teragrep/pth10/steps/teragrep/bloomfilter/FilterTypes.java

```diff
@@ -73,35 +72,33 @@ public FilterTypes(Config config) {
      *
      * @return SortedMap of filter configuration
      */
-    public SortedMap<Long, Double> sortedMap() {
-        final SortedMap<Long, Double> sizesMapFromJson = new TreeMap<>();
-        final Gson gson = new Gson();
-        final List<JsonObject> jsonArray = gson.fromJson(sizesJsonString(), new TypeToken<List<JsonObject>>() {
-        }.getType());
-        for (final JsonObject object : jsonArray) {
-            if (object.has("expected") && object.has("fpp")) {
-                final Long expectedNumOfItems = Long.parseLong(object.get("expected").toString());
-                final Double fpp = Double.parseDouble(object.get("fpp").toString());
-                if (sizesMapFromJson.containsKey(expectedNumOfItems)) {
-                    LOGGER.error("Duplicate value of expected number of items value: <[{}]>", expectedNumOfItems);
-                    throw new RuntimeException("Duplicate entry expected num of items");
-                }
-                sizesMapFromJson.put(expectedNumOfItems, fpp);
-            }
-            else {
-                throw new RuntimeException("JSON did not have expected values of 'expected' or 'fpp'");
-            }
+    public List<FilterField> fieldList() {
+        final List<FilterField> fieldList;
+        try {
+            final Gson gson = new Gson();
+            fieldList = gson.fromJson(sizesJsonString(), new TypeToken<List<FilterField>>() {
+            }.getType());
+        }
+        catch (Exception e) {
```

Comment:
Would it be possible to catch a more specific exception, like JsonSyntaxException (I'm not 100% sure what this fromJson() throws)

Reply:
catching exceptions `JsonIOException` and `JsonSyntaxException` thrown by `fromJson()` now